agent/provider/src/market/negotiator/builtin/manifest.rs

```diff
@@ -25,28 +25,27 @@ impl NegotiatorComponent for ManifestSignature {
             return Ok(NegotiationResult::Ready { offer });
         }
 
-        let manifest = match demand.get_property::<String>(DEMAND_MANIFEST_PROPERTY) {
+        let data = match demand.get_property::<String>(DEMAND_MANIFEST_PROPERTY) {
             Err(Error::NoKey(_)) => return Ok(NegotiationResult::Ready { offer }),
             Err(e) => return rejection(format!("invalid manifest type: {:?}", e)),
-            Ok(s) => match decode_manifest(s) {
+            Ok(s) => match decode_data(s) {
                 Ok(manifest) => manifest,
-                Err(e) => return rejection(format!("invalid manifest: {:?}", e)),
+                Err(e) => return rejection(format!("invalid manifest encoding: {:?}", e)),
             },
         };
 
+        let manifest = decode_manifest(&data)?;
```

Comment:
Probably you want to reject proposal here, not return with error

Reply:
Fixed